smartdashboard/pages/2_Database_Telemetry.py

```diff
@@ -0,0 +1,104 @@
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import os
+import pathlib
+import sys
+import time
+import typing as t
+
+import streamlit as st
+
+from smartdashboard.Experiment_Overview import get_parser
+from smartdashboard.utils.errors import SSDashboardError
+from smartdashboard.utils.ManifestReader import load_manifest
+from smartdashboard.utils.pageSetup import local_css, set_streamlit_page_config
+from smartdashboard.view_builders import db_telem_builder, error_builder
+
+
+def get_path(args: t.List[str]) -> str:
+    """Get the manifest path using the directory
+    path passed in from the command line arguments.
+
+    This will need to be done if a user refreshes
+    the page from the Database Telemetry page. The directory
+    path is used to load in the morchestrators in the manifest.
+
+    :param args: Passed in arguments
+    :type args: t.List[str]
+    :return: Manifest path
+    :rtype: str
+    """
+    arg_parser = get_parser()
+    parsed_args: argparse.Namespace = arg_parser.parse_args(args)
+
+    # default behavior will load a demo manifest from the test samples
+    exp_path = pathlib.Path(__file__).parent.parent.parent
+    manifest_path = exp_path / "tests/utils/manifest_files/manifesttest.json"
+    if parsed_args.directory is not None:
+        exp_path = pathlib.Path(parsed_args.directory)
+        manifest_path = exp_path / ".smartsim/telemetry/manifest.json"
+
+    return str(manifest_path)
+
+
+def build_telemetry_page() -> None:
+    """Build the application components with streamlit
+    for the Database Telemetry page
+    """
+    set_streamlit_page_config()
+
+    curr_path = pathlib.Path(os.path.abspath(__file__)).parent.parent
+    local_css(str(curr_path / "static/style.css"))
+
+    if "manifest" not in st.session_state:
+        manifest_path = get_path(sys.argv[1:])
+        try:
+            manifest = load_manifest(manifest_path)
+            st.session_state["manifest"] = manifest
+        except SSDashboardError as ex:
+            error_builder(ex)
+
+        else:
+            update_telemetry_page()
+
+    else:
+        update_telemetry_page()
+
+
+def update_telemetry_page() -> None:
+    """Update the components for the Database Telemetry page"""
+    views = db_telem_builder(st.session_state["manifest"])
+
+    while True:
+        for v in [views.orc_summary_view, views.memory_view, views.client_view]:
+            v.update()
+
+        time.sleep(1)
+
+
+build_telemetry_page()
```

Comment:
Can we avoid having methods get invoked on import and purposefully call them instead?